app/helpers/sessions_helper.rb

```diff
@@ -6,15 +6,37 @@ def log_in(user)
     session[:cart] = {}
   end
 
+  def remember(user)
+    user.remember
+    cookies.permanent.encrypted[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+    if (user_id = session[:user_id])
+      user = User.find_by(id: user_id)
+      @current_user ||= user if user.session_token == session[:session_token]
+    elsif (user_id = cookies.encrypted[:user_id])
+      user = User.find_by(id: user_id)
```

Comment:
`user = User.find_by(id: user_id)` is present in both cases, why not put it on top and only once?

Reply:
Because the source of the user_id can be different based on the branch of the if, in the first case it comes from the session object, and on the else it comes from the cookies. If I were to put it on top, I would have to make another if for this and I think it would be redundant.

Reply:
good point, thanks 👍🏻